### PR TITLE
Improve query performance when nulls_last option is enabled

### PR DESCRIPTION
**Pull request by natebird**
The separate syntax per database driver allows indices to give performance improvements.

### CODE REVIEW

File: spec/ajax-datatables-rails/datatable/simple_order_spec.rb [hunk 1]
```diff
@@ -28,12 +28,35 @@
...
+        it 'sql query' do
+          expect(nulls_last_order.query('email')).to eq('email DESC IS NULL')
+        end
+      end
+
+      # it 'sql query' do
+      #   expect(nulls_last_order.query('email')).to eq(
+      #     'CASE WHEN email IS NULL THEN 1 ELSE 0 END, email DESC'
+      #   )
+      # end
```
Comment (aantillon) [inline spec/ajax-datatables-rails/datatable/simple_order_spec.rb]: please remove these lines if not needed

**Review by aantillon — CHANGES_REQUESTED** [review]
Awesome!

File: spec/ajax-datatables-rails/datatable/simple_order_spec.rb (hunk 1 above)
Comment (natebird) [reply spec/ajax-datatables-rails/datatable/simple_order_spec.rb]: Good catch. Thanks. I pushed a fix.

**Review by antillas21 — APPROVED** [review]

**Comment by natebird** [comment]
The ENV variable for the database adapter seems to be interfering with the tests I created. Any ideas on how to work around that?

File: lib/ajax-datatables-rails/datatable/simple_order.rb [hunk 2]
```diff
@@ -43,6 +39,19 @@ def sort_nulls_last?
         column.nulls_last? || AjaxDatatablesRails.config.nulls_last == true
       end
 
+      def nulls_last_sql
+        return unless sort_nulls_last?
+
+        case AjaxDatatablesRails.config.db_adapter
+        when -> (a) { a.in?([:pg, :postgresql, :postgres, :oracle]) }
```
Comment (n-rodriguez) [inline lib/ajax-datatables-rails/datatable/simple_order.rb]: Why not :

```ruby
case AjaxDatatablesRails.config.db_adapter
when :pg, :postgresql, :postgres, :oracle
when :mysql, :mysql2, :sqlite, :sqlite3
end
```

**Comment by natebird** [comment]
@antillas21 - can you review this again? It looks like a couple of failed tests due to not finding bundler. Anything you want me to do or is this ready to be merged?

File: lib/ajax-datatables-rails/datatable/simple_order.rb (hunk 2 above)
Comment (natebird) [reply lib/ajax-datatables-rails/datatable/simple_order.rb]: I'm not sure why I did it that way. I updated the case statement. Much better. Thanks!

**Comment by natebird** [comment]
@n-rodriguez Can you look this over again? The Travis failures are related to not finding bundler.

**Pull request merged**